src/CodeAnalysis/Test/MetaAnalyzers/UseReturnValueFromImmutableObjectMethodTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers;
+using Xunit;
+using Microsoft.CodeAnalysis.Analyzers;
+using Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers;
+
+namespace Microsoft.CodeAnalysis.UnitTests.MetaAnalyzers
+{
+    public class UseReturnValueFromImmutableObjectMethodTests : CodeFixTestBase
+    {
+
+        [Fact]
+        public void CSharpVerifyDiagnostics()
+        {
+            var source = @"
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+class TestSimple
+{
+    void M()
+    {
+        Document document = default(Document);
+        document.WithText(default(SourceText));
+
+        Project project = default(Project);
+        project.AddDocument(""Sample.cs"", default(SourceText));
+
+        Solution solution = default(Solution);
+        solution.AddProject(""Sample"", ""Sample"", ""CSharp"");
+    }
+}
+";
+            var documentExpected = GetCSharpExpectedDiagnostic(10, 9);
+            var projectExpected = GetCSharpExpectedDiagnostic(13, 9);
+            var solutionExpected = GetCSharpExpectedDiagnostic(16, 9);
+
+            VerifyCSharp(source, documentExpected, projectExpected, solutionExpected);
+        }
+
+        [Fact]
+        public void CSharp_VerifyDiagnosticOnExtensionMethod()
+        {
+            var source = @"
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+class TestExtensionMethodTrivia
+{
+    void M()
+    {
+        SyntaxNode node = default(SyntaxNode);
+        node.WithLeadingTrivia<SyntaxNode>();
+    }
+}";
+            var expected = GetCSharpExpectedDiagnostic(10, 9);
+            VerifyCSharp(source, expected);
+        }
+
+
+        protected override CodeFixProvider GetBasicCodeFixProvider()
+        {
+            return null;
+        }
+
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        {
+            return null;
+        }
+
+        protected override CodeFixProvider GetCSharpCodeFixProvider()
+        {
+            return null;
+        }
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new CSharpImmutableObjectMethodAnalyzer();
+        }
+
+        private static DiagnosticResult GetCSharpExpectedDiagnostic(int line, int column)
+        {
+            return GetExpectedDiagnostic(LanguageNames.CSharp, line, column);
+        }
+
+        private static DiagnosticResult GetExpectedDiagnostic(string language, int line, int column)
+        {
+            var fileName = language == LanguageNames.CSharp ? "Test0.cs" : "Test0.vb";
+            return new DiagnosticResult
+            {
+                Id = DiagnosticIds.DoNotIgnoreReturnValueOnImmutableObjectMethodInvocation,
+                Message = CodeAnalysisDiagnosticsResources.DoNotIgnoreReturnValueOnImmutableObjectMethodInvocationMessage,
```

Comment:
Right now my tests are failing because of this message. 

My message is dynamically created based on the method the user is trying to use. It's something similar to:

```
'MethodDeclarationSyntax' is immutable and 'WithBody' will not have any effect on it. Consider using the return value from 'WithBody'.
```

I noticed while building my tests that all other diagnostics seem to use static messages that can be compared easier in tests. I like being able to tell the user exactly what type is immutable and what method invocation is causing the issue. Is there a way to make this approach work with this project's testing suite?